tests/bsim/bluetooth/host/misc/sample_test/src/main.c

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+
+#include "bs_tracing.h"
+#include "bstests.h"
+
+extern void entrypoint_dut(void);
+extern void entrypoint_peer(void);
+extern enum bst_result_t bst_result;
+
+
+static void test_tick(bs_time_t HW_device_time)
+{
+	/* This callback will fire right before the simulation times out.
```

Comment:
Maybe this should be corrected to say that it will be called with whatever deadline you set, which in this example is set by -test_deadline.